app/actions/remote/thread.ts

```diff
@@ -368,39 +381,56 @@ export const syncTeamThreads = async (
                 return {error: allUnreadThreads.error || latestThreads.error};
             }
 
-            const dedupe = new Set(latestThreads.threads?.map((t) => t.id));
-
             if (latestThreads.threads?.length) {
                 // We are fetching the threads for the first time. We get "latest" and "earliest" values.
+                // At this point we may receive threads without replies, so we also check the post.create_at timestamp.
                 const {earliestThread, latestThread} = getThreadsListEdges(latestThreads.threads);
-                syncDataUpdate.latest = latestThread.last_reply_at;
-                syncDataUpdate.earliest = earliestThread.last_reply_at;
+                syncDataUpdate.latest = latestThread.last_reply_at || latestThread.post.create_at;
+                syncDataUpdate.earliest = earliestThread.last_reply_at || earliestThread.post.create_at;
 
                 threads.push(...latestThreads.threads);
             }
             if (allUnreadThreads.threads?.length) {
+                const dedupe = new Set(latestThreads.threads?.map((t) => t.id));
                 const unread = allUnreadThreads.threads.filter((u) => !dedupe.has(u.id));
                 threads.push(...unread);
             }
         } else {
-            const allNewThreads = await fetchThreads(
-                serverUrl,
-                teamId,
-                {deleted: true, since: syncData.latest + 1, excludeDirect},
-                undefined,
-                undefined,
-                groupLabel,
-            );
+            const [allUnreadThreads, allNewThreads] = await Promise.all([
+                fetchThreads(
+                    serverUrl,
+                    teamId,
+                    {unread: true, excludeDirect},
+                    Direction.Down,
+                    undefined,
+                    groupLabel,
+                ),
+                fetchThreads(
+                    serverUrl,
+                    teamId,
+                    {deleted: true, since: refresh ? undefined : syncData.latest + 1, excludeDirect},
+                    undefined,
+                    1,
+                    groupLabel,
+                ),
+            ]);
+
             if (allNewThreads.error) {
                 return {error: allNewThreads.error};
             }
             if (allNewThreads.threads?.length) {
                 // As we are syncing, we get all new threads and we will update the "latest" value.
                 const {latestThread} = getThreadsListEdges(allNewThreads.threads);
-                syncDataUpdate.latest = latestThread.last_reply_at;
+                const latestDate = latestThread.last_reply_at || latestThread.post.create_at;
```

Comment:
How come we don't do latestThread.post.lastReplyAt like we did in transformer/thread.ts?

```
thread.lastReplyAt = raw.last_reply_at || record?.lastReplyAt || raw.post.create_at;
```

Reply:
The piece of code you pasted says:

- Give me the new last reply at
- If it is 0, give me whatever reply at I already had
- If that is still 0, get me the thread post create at

The code you are referencing is doesn't deal with what we used to have, so it only checks:
- Get the last reply at
- If it is 0, give me the create at of the post

Reply:
There is no post last reply at, there is only thread last reply at.